CleanWeather/Scenes/FavouriteCities/FavouriteCitiesController.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  FavouriteCitiesController.swift
+//  CleanWeather
+//
+//  Created by Marek Skrzelowski on 04/12/2019.
+//  Copyright © 2019 vandermesis. All rights reserved.
+//
+
+import UIKit
+
+protocol FavouriteCitiesPresentable: SpinnerPresentable & AlertPresentable {
+    func displayCities(_ city: [FavouriteCitiesListDisplayable])
+}
+
+final class FavouriteCitiesController: SharedViewController {
+
+    @IBOutlet private weak var citiesTableView: UITableView!
+
+    private let interactor: FavouriteCitiesInteractor
+
+    private var citiesDataSource = [FavouriteCitiesListDisplayable]()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupTableView()
+        setupNavigationBar()
+        interactor.getCities()
+    }
+
+    init(interactor: FavouriteCitiesInteractor) {
+        self.interactor = interactor
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    deinit {
+        print("FavouriteCitiesController DEALLOCATED")
```

Comment:
jak tego nie uzywasz, to bym wywalil te printy, bo tylko smietnik robią
polecam ci raz na jakis czas odpalic memory viewer i patrzec czy sceny zwalniaja pamiec jak z nich wychodzisz

Reply:
i po drugie, jak masz inity to nie rozdzielaj ich innymi metodami, 
teraz masz

init
deinit
required init

Bardzo źle :D